code_challenges/adhoc/ad_hock_challenge.py

```diff
@@ -0,0 +1,32 @@
+#Write a function that takes a dictionary, string, and iterable of values.
+# The dictionaries keys will be strings and the values will be functions.
+# The return value should be the return value from the function from the input dict,  at the input string, 
+# when called with the values in the input iterable as arguments.
+def find_the_function(functions, function_name, the_numbers):
```

Comment:
`the_numbers` parameter name makes an extra assumption about the inputs that wasn't stated.  `inputs` or `values` would be better since we don't know what types they will be.